pandas/io/formats/xml.py

```diff
@@ -311,16 +342,15 @@ def build_tree(self) -> bytes:
         )
 
         for d in self.frame_dicts.values():
-            self.d = d
-            self.elem_row = SubElement(self.root, f"{self.prefix_uri}{self.row_name}")
+            elem_row = SubElement(self.root, f"{self.prefix_uri}{self.row_name}")
 
             if not self.attr_cols and not self.elem_cols:
-                self.elem_cols = list(self.d.keys())
-                self.build_elems()
+                self.elem_cols = list(d.keys())
+                self.build_elems(d, elem_row)
```

Comment:
should this not assign?

Reply:
`build_elems()` returns None. XML trees are built in place from root with elements added to it. After processing, the root (not any sub element) is passed to output.